### PR TITLE
ESSNTL-4070: add '[not implemented]' to group paths

### PR DESCRIPTION
**Pull request by FabriciaDinizRH**
# Overview

This PR is being created to address [ESSNTL-4070](https://issues.redhat.com/browse/ESSNTL-4070).
Adds `[not implemented]` to group paths to prevent the paths being used by customers. 

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

**Review by kruai — DISMISSED** [review]

File: swagger/api.spec.yaml [hunk 1]
```diff
@@ -349,9 +349,9 @@ paths:
       operationId: api.group.get_group_list
       tags:
         - groups
-      summary: Read the entire list of groups
+      summary: Read the entire list of groups [Not implemented]
```
Comment (strider) [inline swagger/api.spec.yaml:352]: Nitpicking here:

`s/Not implemented/Not Implemented/g`

**Review by strider — APPROVED** [review]

**Review by strider — APPROVED** [review]
Thanks @FabriciaDinizRH!

**Comment by FabriciaDinizRH** [comment]
@strider all good

**Comment by strider** [comment]
Let's test @fstavela's CI fix and push this one forward! ;-)

**Pull request merged**